trellis-api/src/main/java/org/trellisldp/api/ResourceService.java

```diff
@@ -56,7 +66,37 @@
      * @param dataset the dataset
      * @return whether the resource was added
      */
-    Future<Boolean> put(IRI identifier, IRI ixnModel, Dataset dataset);
+    Future<Boolean> create(IRI identifier, IRI ixnModel, Dataset dataset);
+
+    @Override
+    default Future<Boolean> replace(IRI id, Resource res) {
+        return replace(id, res.getInteractionModel(), res.dataset());
+    }
+
+    /**
+     * Replace a resource in the server.
+     *
+     * @param identifier the identifier for the new resource
+     * @param ixnModel the LDP interaction model for this resource
+     * @param dataset the dataset
+     * @return whether the resource was replaced
+     */
+    Future<Boolean> replace(IRI identifier, IRI ixnModel, Dataset dataset);
+
+    @Override
+    default Future<Boolean> delete(IRI id, Resource res) {
+        return delete(id, res.getInteractionModel(), res.dataset());
+    }
+
+    /**
+     * Delete a resource from the server.
+     *
+     * @param identifier the identifier for the new resource
+     * @param ixnModel the new LDP interaction model for this resource
+     * @param dataset the dataset
+     * @return whether the resource was deleted
+     */
+    Future<Boolean> delete(IRI identifier, IRI ixnModel, Dataset dataset);
```

Comment:
I have been wondering if this method shouldn't be just `delete(IRI)`. What do the `ixnModel` and `dataset` mean in this context? This would clearly have implications for other interfaces.

Reply:
The problem is that deleting something doesn't. It tombstones it, and we need to know with what to build the tombstone. Unless you are saying that from the HTTP POV, a `DELETE` message should use `::replace` on the backend? (Which actually makes a lot of sense.)

Reply:
We would still want to distinguish between an empty `HTTP PUT` (one that removes all triples) and a `HTTP DELETE` (one that removes triples and sets a tombstone -- though, a persistence impl could choose to really delete the resource, sans tombstone).

At present, the Triplestore impl just ignores the `ixnModel` and `dataset` values, but that also doesn't mean that's what every impl needs to do. I'm fine with leaving this as-is because an implementation still has a lot of control over these methods.

Reply:
Yeah, the thing is to both distinguish between `PUT` and `DELETE` and also to distinguish between `::replace` and `::delete`. Can we say for sure that any HTTP `PUT` will result in a `::replace` and any HTTP `DELETE` will result in a `::delete`? If so, we need to leave these params in. If both `PUT` and `DELETE` will map to `::replace` (with `DELETE` just becoming a special case of "replace with a tombstone") then we can probably drop the params.

Reply:
I am actually noticing a condition that speaks exactly to your point: https://github.com/trellis-ldp/trellis/blob/FinishNewPersistence/trellis-http/src/main/java/org/trellisldp/http/impl/DeleteHandler.java#L92-L101

When an ACL graph is deleted (via DELETE), the user-managed graph remains intact. Furthermore, the `baseURL` value needs to be sent along for use with the event stream. (This also means that there is a bug in the triplestore impl). Either way, the signature definitely needs to stay the way it is.

Reply:
Okay, that works for me! On to some tests and some Git-fu.